go/vt/schemadiff/schema_diff_test.go

```diff
@@ -0,0 +1,657 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schemadiff
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPermutations(t *testing.T) {
+	var (
+		fromQueries = []string{
+			"create table t1 (id int primary key, info int not null);",
+			"create table t2 (id int primary key, ts timestamp);",
+			"create view v1 as select id from t1",
+		}
+		toQueries = []string{
+			"create table t1 (id int primary key, info int not null, i int);",
+			"create table t2 (id int primary key);",
+			"create view v1 as select id, info from t1",
+			"create view v2 as select id from t2",
+		}
+	)
```

Comment:
I feel like we could probably benefit from additional test cases? Maybe not though. Your call.

Reply:
Added multiple tests.